src/index.ts

```diff
@@ -41,6 +46,23 @@ export class GraphQLClient {
       throw new ClientError({ ...errorResult, status: response.status}, {query, variables})
     }
   }
+
+  setHeaders (headers: Headers): GraphQLClient {
+    this.options.headers = headers
+
+    return this
+  }
+
+  setHeader (key: string, value: string): GraphQLClient {
+    const { headers } = this.options
+
+    if (headers) {
+      headers[key] = value
+    } else {
+      Object.assign(this.options, { headers: { [key]: value } })
```

Comment:
Any reason you don't write:

```ts
this.options.headers = { [key]: value }
```

Reads a lot easier.

Reply:
because i think there are two place use headers,
```
if (this.options.headers) {
  this.options.headers = { [key]: value }
}
```
so i think write like this will visit headers twice 😂

Reply:
Sorry, I cannot follow. Maybe someone else can explain?

Reply:
so you mean
```
if (headers) {
  this.options.headers = { [key]: value }
}
```
sorry, i agree with your opinion about readability, and i'm not sure about my reason , so i ask whether visiting an attribution twice will cause a little bit of performance loss .

Reply:
I'd definitely go in favour of readability in this case.

Reply:
thanks. i get your point, forgive me that i read your advise wrong before